extensions/multiline.md

```diff
@@ -0,0 +1,241 @@
+
+Clients MUST ignore every token with a key that they don't understand.
+
+The only defined capability key so far is:
+
+* `max-bytes` - This defines the maximum allowed total byte length of multiline batched content *REQUIRED*
+* `max-lines` - This defines the maximum allowed number of lines in a multiline batch content *OPTIONAL*
+
+### Tags
+
+This specification adds the `draft/multiline-concat` message tag, which has no value and is used to send message lines that extend beyond the usual byte length limit. Its usage is described below.
+
+### Batch types
+
+This specification adds the `draft/multiline` batch type, and introduces client initiated batches. These use the same syntax as server initiated batches.
+
+In addition to the base batch parameters (reference-tag and type) a multiline batch has one additional parameter, the target recipient.
+
+Multiline batches MUST contain one or more PRIVMSG lines whose target MUST match the batch target.
+
+When receiving a well-formed mulitiline message batch, implementations MUST collect each PRIVMSG message content and wait until the full batch has been received before processing the message. Processing in this context refers to:
+
+* Servers: delivering the batch to the intended recipients
+* Clients: displaying the batched message to the user
+
+Messages in a multiline batch MUST be concatenated with a line break unless the `draft/multiline-concat` message tag is sent, in which case the message is directly concatenated with the previous message without any separation.
+
+Clients MUST NOT send messages other than PRIVMSG while a multiline batch is open.
+
+### Fallback
+
+When delivering multiline batches to clients that have not negotiated the multiline capability, servers MUST deliver the component messages without using a multiline BATCH.
+
+Any tags that would have been added to the batch, e.g. message IDs, account tags etc MUST be included on the first message line. Tags MAY also be included on subsequent lines where it makes sense to do so.
+
+### Errors
+
+Servers MUST use `FAIL` messages from the [standard replies][] framework to notify clients of errors with multiline batches. The command is `BATCH` and the following codes are defined:
+
+* `MULTILINE_MAX_BYTES <limit>`: `max-bytes` limit exceeded
+* `MULTILINE_MAX_LINES <limit>`: `max-lines` limit exceeded
+* `MULTILINE_INVALID_TARGET <batch-target> <provided-target>`: mismatched PRIVMSG target within a batch
+* `MULTILINE_INVALID`: any other error
+
+## Interaction with other specifications
+
+### Message tags ([spec][message tags])
+
+Clients MUST only send non-`batch` tags on the opening BATCH command when sending multiline batches.
```

Comment:
Suggested language from discussion in #ircv3: Clients MUST NOT send tags other than draft/multiline-concat on messages within the batch. In particular, all client-only tags associated with the message must be sent attached to the initial BATCH message.